### PR TITLE
IS-8545: added docs on how rescan works

### PR DESCRIPTION
**Pull request by knutj42**

### CODE REVIEW

File: configuration.rst [hunk 1]
```diff
@@ -8362,3 +8365,89 @@ A scheduled pump running every 5 minutes from 14:00 and ending at 14:55, AND fir
...
+
+   In this case the incremental run will use the old version of the merge-source, which may produce erronous results.
+   The entities from the incremental run will not be put into the sink's seen-tracker. The incremental run will not
+   overwrite any entities that have been produced by the rescan run.
+
+   Once the rescan finishes, any incremental run in progress will be stopped. The rescan will then process any entities
+   that have appeared since the start of the rescan. Once that is done, the rescan will do deletion-tracking. This will
+   delete any erronous entities that was emitted by the incremental run.
+
+   Caveats of doing rescan+incremental runs:
```
Comment (tombech) [inline configuration.rst:8434]: Is this a repeat of the title above?

File: configuration.rst [hunk 2]
```diff
@@ -8362,3 +8365,89 @@ A scheduled pump running every 5 minutes from 14:00 and ending at 14:55, AND fir
...
+
+Only one incremental run can be active at once, but once an incremental run has finished a new incremental run can be
+started. A rescan run can also be started while an incremental run is in progress.
+
+The incremental runs will not do retries, since the rescan will reprocess any previously failed entities.
+The incremental runs will do dependency tracking.
+
+
+
+Caveats of doing rescan+incremental runs:
```
Comment (tombech) [inline configuration.rst:8448]: Repeat?

Comment (knutj42) [reply configuration.rst]: yep, this was a repeat

File: configuration.rst (hunk 1 above)
Comment (knutj42) [reply configuration.rst:8434]: Not really, there is one unique "Caveat.."-section for each "flavor" of rescan.

File: configuration.rst [hunk 3]
```diff
@@ -8362,3 +8365,80 @@ A scheduled pump running every 5 minutes from 14:00 and ending at 14:55, AND fir
...
+
+   * The order of the resulting entities can be different that it would be in a normal "reset"-run.
+   * The output can temporarily contain erronous entities (produced by the incremental runs).
+     Such entities will deleted once the rescan has finished.
+
+Only one incremental run can be active at once, but once an incremental run has finished a new incremental run can be
+started. A rescan run can also be started while an incremental run is in progress.
+
+The incremental runs will not do retries, since the rescan will reprocess any previously failed entities.
+The incremental runs will do dependency tracking.
```
Comment (knutj42) [inline configuration.rst:8444]: Add docs on the global default setting and the pipe-setting for enabling background rescans.

**Review by tombech — APPROVED** [review]

File: configuration.rst [hunk 4]
```diff
@@ -8395,7 +8413,16 @@ Incremental run:
 Rescan:
   This is what a pump does when it is started by the :ref:`rescan_cron_expression <pump_rescan_cron_expression>` or
   :ref:`rescan_run_count <pump_rescan_run_count>` config-properties (or if it is manually started by the
-  "rescan" pump-operation). It will process all the source-entities, and do deletion tracking when finished.
+  "start-rescan" pump-operation). It will process all the source-entities, and do deletion tracking when finished.
+
+  Only pipes with a :ref:`dataset sink <dataset_sink>` supports rescans.
```
Comment (grove) [inline configuration.rst:8418]: You mean *background rescans*, or?

Comment (knutj42) [reply configuration.rst]: yep, fixed.

**Pull request merged**